src/pages/confirmation.tsx

```diff
@@ -0,0 +1,104 @@
+                    <div className="govuk-grid-column-one-thirds">
+                        <h1 className="govuk-heading-l">Check your answers before sending your fares information</h1>
+                        <ConfirmationTable
+                            confirmationElements={[
+                                {
+                                    header: 'Fare Information',
+                                    innerElements: [
+                                        { name: 'Faretype', content: startCase(fareType), href: '/fareType' },
+                                        {
+                                            name: 'Passenger Type',
+                                            content: startCase(passengerType),
+                                            href: '/passengerType',
+                                        },
+                                        {
+                                            name: 'Time Restrictions',
+                                            content: timeRestrictions,
+                                            href: '/timeRestrictions',
+                                        },
+                                    ],
+                                },
+                            ]}
+                        />
+                        <h2 className="govuk-heading-m">Now send your fares information</h2>
+
+                        <p className="govuk-body">
+                            By submitting this notification you are confirming that, to the best of your knowledge, the
+                            details you are providing are correct.
+                        </p>
+                    </div>
+                </div>
+                <input type="submit" value="Continue" id="continue-button" className="govuk-button" />
+            </>
+        </CsrfForm>
+    </TwoThirdsLayout>
+);
+
+export const getServerSideProps = (ctx: NextPageContextWithSession): { props: ConfirmationProps } => {
+    const fareTypeInfo = getSessionAttribute(ctx.req, FARE_TYPE_ATTRIBUTE);
+    const passengerTypeInfo = getSessionAttribute(ctx.req, PASSENGER_TYPE_ATTRIBUTE);
+    const timeRestrictionsInfo = getSessionAttribute(ctx.req, TIME_RESTRICTIONS_DEFINITION_ATTRIBUTE);
+    let finalPassengerInfo = '';
+    let finalFareTypeInfo = '';
+
+    if (passengerTypeInfo) {
+        if (isPassengerTypeAttributeWithErrors(passengerTypeInfo)) {
+            throw new Error('User has reached confirmation page with incorrect passenger type info.');
+        }
+        finalPassengerInfo = passengerTypeInfo.passengerType;
+    }
+
+    if (!fareTypeInfo || isFareTypeAttributeWithErrors(fareTypeInfo)) {
+        throw new Error('User has reached confirmation page without necessary data');
+    }
+
+    if (timeRestrictionsInfo && isTimeRestrictionsDefinitionWithErrors(timeRestrictionsInfo)) {
+        throw new Error('User has reached confirmation page without necessary data');
+    }
```

Comment:
I know we're coming back to finish this work, but would be nicer to lump these errors together for future reference